Realm/ObjectServerTests/RLMObjectServerTests.m

```diff
@@ -88,97 +113,206 @@ - (void)testOpenRealmWithAdminToken {
     if (self.isParent) {
-        [user waitForDownloadToFinish:url];
         // Add objects.
-        [r beginWriteTransaction];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-1"]]];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-2"]]];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-3"]]];
-        [r commitWriteTransaction];
-        [user waitForUploadToFinish:url];
+        [self addSyncObjectsToRealm:r descriptions:@[@"parent-1", @"parent-2", @"parent-3"]];
+        WAIT_FOR_UPLOAD(user, url);
         CHECK_COUNT(3, SyncObject, r);
         RLMRunChildAndWait();
-        [user waitForDownloadToFinish:url];
+        WAIT_FOR_DOWNLOAD(user, url);
         CHECK_COUNT(0, SyncObject, r);
     } else {
-        [user waitForDownloadToFinish:url];
+        WAIT_FOR_DOWNLOAD(user, url);
         CHECK_COUNT(3, SyncObject, r);
         [r beginWriteTransaction];
         [r deleteAllObjects];
         [r commitWriteTransaction];
-        [user waitForUploadToFinish:url];
+        WAIT_FOR_UPLOAD(user, url);
         CHECK_COUNT(0, SyncObject, r);
     }
 }
 
+#pragma mark - Multiple Realm Sync
+
+/// If a client opens multiple Realms, there should be one session object for each Realm that was opened.
+- (void)testMultipleRealmsSessions {
+    NSURL *urlA = CUSTOM_REALM_URL(@"a");
+    NSURL *urlB = CUSTOM_REALM_URL(@"b");
+    NSURL *urlC = CUSTOM_REALM_URL(@"c");
+    RLMSyncUser *user = [self logInUserForCredential:[RLMObjectServerTests basicCredentialWithName:ACCOUNT_NAME()
+                                                                                     createAccount:self.isParent]
+                                              server:[RLMObjectServerTests authServerURL]];
+    // Open three Realms.
+    __attribute__((objc_precise_lifetime)) RLMRealm *realmA = [self openRealmForURL:urlA user:user];
+    __attribute__((objc_precise_lifetime)) RLMRealm *realmB = [self openRealmForURL:urlB user:user];
+    __attribute__((objc_precise_lifetime)) RLMRealm *realmC = [self openRealmForURL:urlC user:user];
+    // Make sure there are three active sessions for the user.
+    XCTAssert(user.allSessions.count == 3, @"Expected 3 sessions, but didn't get 3 sessions");
+    XCTAssertNotNil([user sessionForURL:urlA], @"Expected to get a session for URL A");
+    XCTAssertNotNil([user sessionForURL:urlB], @"Expected to get a session for URL B");
+    XCTAssertNotNil([user sessionForURL:urlC], @"Expected to get a session for URL C");
+    XCTAssertTrue([user sessionForURL:urlA].state == RLMSyncSessionStateActive, @"Expected active session for URL A");
+    XCTAssertTrue([user sessionForURL:urlB].state == RLMSyncSessionStateActive, @"Expected active session for URL B");
+    XCTAssertTrue([user sessionForURL:urlC].state == RLMSyncSessionStateActive, @"Expected active session for URL C");
+}
+/*
+/// A client should be able to open multiple Realms and add objects to each of them.
```

Comment:
I'd prefer prefixing disabled tests with `DISABLED_` rather than commenting them out to at least ensure they compile.